src/main/java/core/basesyntax/domain/Fruit.java

```diff
@@ -0,0 +1,73 @@
+package core.basesyntax.domain;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Fruit {
+    private Operation operation;
+    private String name;
+    private int quantity;
+
+    public Fruit(String name, int quantity) {
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public Fruit(Operation operation, String name, int quantity) {
+        this.operation = operation;
+        this.name = name;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Fruit fruit = (Fruit) o;
+        return quantity == fruit.quantity
+                && operation == fruit.operation
+                && Objects.equals(name, fruit.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(operation, name, quantity);
+    }
+
+    public enum Operation {
```

Comment:
The 'Operation' enum is placed inside the 'Fruit' class. According to the checklist, we should not keep all logic in a single package or class. Please move the 'Operation' enum to its own file in the appropriate package, possibly within a 'strategy' package if you are implementing the Strategy pattern.